### PR TITLE
apply proxy from outside

### PR DESCRIPTION
**Pull request by hannesa2**
TODO We should read values from `.env` file or '.env_file'
OR alternative redirect host environment variables  directly to docker
`docker build --build-arg PROXY_PORT=$HOST_PROXY_PORT` but for `docker-compose.yml` I didn't found a way now

### CODE REVIEW

**Comment by hannesa2** [comment]
Generell question: The proxy issue is it just during build or in image/runtime as well ?

**Comment by hannesa2** [comment]
Currently I only see the way to use a new `build.sh` with 
`docker build --build-arg PROXY_PORT=$HOST_PROXY_PORT` (and don't use `docker-compose.yml`)

any better idea ?

**Comment by originx** [comment]
We can have it with docker compose of course it is cleaner for example:

android_emulator_image:
  environment:
    - http_proxy

This will take the env variable http_proxy from shell where you run docker-compose and pass it to the container

Or you can specify the env file and use a script build.sh or start.sh to prepare and start everything. 

https://docs.docker.com/compose/environment-variables/#pass-environment-variables-to-containers

**Comment by hannesa2** [comment]
Yes, to the container ! But I was not able to read it during _build_.
In my tests only ARG were available during build

File: Dockerfile [hunk 1]
```diff
@@ -4,9 +4,20 @@
...
+## ARG's for build time with default values
+# TODO remove this comment. Set APROXY_PORT with an error. It's the fastest way to see if it's work
+ARG APROXY_PROTOCOL=http
+ARG APROXY_SERVER=localhost
+ARG APROXY_PORT=80
+
+# ENV's are for image, probably we need ot later
+ENV EPROXY_PROTOCOL=$APROXY_PROTOCOL
+ENV EPROXY_SERVER=$APROXY_SERVER
+ENV EPROXY_PORT=$APROXY_PORT
```
Comment (hannesa2) [inline Dockerfile]: remove this line and use in L49 EPROXY_PORT you will have no value

File: docker-compose.yml [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+    build:
+       context: .
+       args:
+         APROXY_PROTOCOL : httpA
+         APROXY_SERVER : localhostA
+         APROXY_PORT : 8080A
     ports:
       - "5554:5554"
+    environment:
+      - EPROXY_PORT=9999E
```
Comment (hannesa2) [inline docker-compose.yml]: the expectation is this value, but it's "" in L49

**Comment by hannesa2** [comment]
After several tries the error
`Warning: An error occurred while preparing SDK package Android Support Repository: SSL peer shut down incorrectly.` disappeared.

In my point of view it's done currently and be able to merge

**Comment by hannesa2** [comment]
you look not in last commit, the last (without WIP prefix) removes the `--no-cache`.
It was helpful to develop the proxy

**Comment by originx** [comment]
sadly when running on a system without proxy set this does not work, for example this would happen:
 ENV http_proxy http://$APROXY_SERVER:$APROXY_PORT

File: Dockerfile [hunk 3]
```diff
@@ -16,6 +26,11 @@ EXPOSE 5900
 # Use baseimage-docker's init system.
 CMD ["/sbin/my_init"]
 
+RUN echo Proxy setting: $APROXY_PROTOCOL $APROXY_SERVER $APROXY_PORT
```
Comment (originx) [inline Dockerfile]: this is not needed, please remove and just move setting of the http_proxy environment variable above the apt-get commands.

`#system respected proxy env variables, needs to be set before apt-get commands
ENV http_proxy http://$APROXY_SERVER:$APROXY_PORT
ENV https_proxy http://$APROXY_SERVER:$APROXY_PORT
RUN echo http_proxy=$http_proxy`

before you run any apt-get install and apt-get will respect these variables then

File: Dockerfile [hunk 4]
```diff
@@ -16,6 +26,11 @@ EXPOSE 5900
...
+RUN echo "Acquire::http::Proxy \"http_proxy=$APROXY_PROTOCOL://$APROXY_SERVER:$APROXY_PORT\";" > /etc/apt/apt.conf.d/30proxy
```
Comment (originx) [inline Dockerfile]: remove, see above comment

File: Dockerfile [hunk 5]
```diff
@@ -27,31 +42,35 @@ ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64
...
+RUN yes | sdkmanager "extras;android;m2repository" --verbose --proxy_host=$APROXY_SERVER --proxy_port=$APROXY_PORT --proxy=$APROXY_PROTOCOL
+RUN yes | sdkmanager "extras;google;m2repository" --verbose --proxy_host=$APROXY_SERVER --proxy_port=$APROXY_PORT --proxy=$APROXY_PROTOCOL
 
 # Add platform-tools and emulator to path
 ENV PATH $PATH:$ANDROID_HOME/platform-tools
 ENV PATH $PATH:$ANDROID_HOME/emulator
 
-#Install latest android emulator   system images
+#Install latest android emulator system images
 ENV EMULATOR_IMAGE "system-images;android-25;google_apis;armeabi-v7a"
```
Comment (originx) [inline Dockerfile]: later on we can make these things configurable as parameters just like proxy, but not in this PR

File: docker-compose.yml [hunk 6]
```diff
@@ -1,6 +1,17 @@
 version: "3.6"
 services:
   android_emulator:
-    build: .
+    build:
+       context: .
+       args:
+         APROXY_PROTOCOL : httpA
```
Comment (originx) [inline docker-compose.yml]: we can remove it from here if you pass it via build arguments?

**Comment by hannesa2** [comment]
Exactly, (I mention it yesterday) a system without proxy will fail. Here a bridge with/out proxy is missing

File: Dockerfile (hunk 3 above)
Comment (hannesa2) [reply Dockerfile]: fyi
`ENV http_proxy ...` doesn't help.

This line makes the magic: set proxy system wide:
`RUN echo "Acquire::http::Proxy \"http_proxy=$APROXY_PROTOCOL://$APROXY_SERVER:$APROXY_PORT\";" > /etc/apt/apt.conf.d/30proxy`

**Pull request closed without merging**